### DISCUSSION TITLE
Restrict database schema changes in web UI

### DISCUSSION THREAD

**Discussion by Magnuti** [supabase/supabase#20738] [Feature Requests]
In my team we are developing locally using migrations. These migrations are added to Git and we use GitHub Actions to deploy the migrations to our Supabase instance. This workflow works excellent for us as it is robust and allows us to track database changes.

However, we find it "too easy" to accidentally modify the database schema in our remote Supabase instance. Since we use the web UI locally to make database schema changes, once or twice we have accidentally changed our remote Supabase instance instead of our local Supabase instance.

Could it be possible to disable the "Insert column" button in settings, so that the only way to change the schema is through migrations or by SQL queries?

<img width="254" alt="image" src="https://github.com/supabase/supabase/assets/43852515/da4bba6f-156d-4f74-93f2-08127aafd227">

This is especially important for production instances, where we often want to view or update data, but we never want to make schema changes directly in production.

**Comment by stefan-girlich** [comment]
Same issue here; see also [this similar discussion](https://github.com/orgs/supabase/discussions/21771).
My organization is currently looking into using Supabase, but the fact that prod changes can be applied outside the CI/CD process is a blocker for us.

**Comment by gwesterman** [comment]
Please let us prohibit making changes to the database via the GUI.
